samza-operator/src/main/java/org/apache/samza/operators/impl/SessionWindowOperatorImpl.java

```diff
@@ -16,50 +16,52 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.samza.operators.impl.window;
+package org.apache.samza.operators.impl;
 
 import org.apache.samza.operators.MessageStream;
-import org.apache.samza.operators.WindowState;
+import org.apache.samza.operators.StateStoreImpl;
+import org.apache.samza.operators.spec.WindowOperatorSpec;
+import org.apache.samza.operators.windows.WindowState;
 import org.apache.samza.operators.data.Message;
-import org.apache.samza.operators.internal.Operators.WindowOperator;
-import org.apache.samza.operators.internal.WindowOutput;
-import org.apache.samza.operators.impl.OperatorImpl;
-import org.apache.samza.operators.impl.StateStoreImpl;
+import org.apache.samza.operators.windows.WindowOutput;
 import org.apache.samza.storage.kv.Entry;
 import org.apache.samza.task.MessageCollector;
 import org.apache.samza.task.TaskContext;
 import org.apache.samza.task.TaskCoordinator;
 
 
 /**
- * Default implementation class of a {@link WindowOperator} for a session window.
+ * Default implementation class of a {@link WindowOperatorSpec} for a session window.
  *
  * @param <M>  the type of input {@link Message}
  * @param <RK>  the type of window key
  * @param <WS>  the type of window state
  * @param <RM>  the type of aggregated value of the window
  */
-public class SessionWindowImpl<M extends Message, RK, WS extends WindowState, RM extends WindowOutput<RK, ?>> extends
-    OperatorImpl<M, RM> {
-  private final WindowOperator<M, RK, WS, RM> sessWnd;
-  private StateStoreImpl<M, RK, WS> wndStore = null;
+class SessionWindowOperatorImpl<M extends Message, RK, WS extends WindowState, RM extends WindowOutput<RK, ?>>
+    extends OperatorImpl<M, RM> {
 
-  public SessionWindowImpl(WindowOperator<M, RK, WS, RM> sessWnd) {
-    this.sessWnd = sessWnd;
+  private final WindowOperatorSpec<M, RK, WS, RM> windowSpec;
+  private StateStoreImpl<M, RK, WS> stateStore = null;
+
+  SessionWindowOperatorImpl(WindowOperatorSpec<M, RK, WS, RM> windowSpec) {
+    this.windowSpec = windowSpec;
   }
 
-  @Override protected void onNext(M message, MessageCollector collector, TaskCoordinator coordinator) {
-    Entry<RK, WS> state = this.wndStore.getState(message);
-    this.nextProcessors(this.sessWnd.getFunction().apply(message, state), collector, coordinator);
-    this.wndStore.updateState(message, state);
+  @Override
+  public void init(MessageStream<M> source, TaskContext context) {
```

Comment:
This method can be completely removed if source and context are passed in as parameters to the constructor.